src/include/traffic_cop/portal.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <sqlite3.h>
+#include <memory>
+#include <vector>
+#include "type/transient_value.h"
+
+namespace terrier::traffic_cop {
+
+/**
+ * A portal is a statement with bound parameters and is ready to execute.
+ */
+struct Portal {
```

Comment:
Will the real system have the same concept? Is this not just `ExecutionContext`?